apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.js

```diff
@@ -40,25 +62,39 @@ router.get(
       throw error.make(403, 'Access denied (must be a student data viewer)');
     }
     res.locals.logCsvFilename = logCsvFilename(res.locals);
-    res.locals.assessment_instance_stats = (
-      await sqldb.queryAsync(sql.assessment_instance_stats, {
+    res.locals.assessment_instance_stats = await sqldb.queryRows(
+      sql.assessment_instance_stats,
+      {
         assessment_instance_id: res.locals.assessment_instance.id,
-      })
-    ).rows;
+      },
+      AssessementInstanceStatsSchema,
+    );
 
-    const dateDurationResult = await sqldb.queryOneRowAsync(sql.select_date_formatted_duration, {
-      assessment_instance_id: res.locals.assessment_instance.id,
-    });
+    const dateDurationResult = await sqldb.queryRow(
+      sql.select_date_formatted_duration,
+      {
+        assessment_instance_id: res.locals.assessment_instance.id,
+      },
+      z.object({
+        assessment_instance_date_formatted: z.string(),
+        assessment_instance_duration: z.string(),
+      }),
```

Comment:
Nit: let's declare all schemas in the top-level scope instead of creating new ones for each request.

Reply:
I have now moved schemas to the top-level scope in commit [ac859c4](https://github.com/PrairieLearn/PrairieLearn/pull/9072/commits/ac859c49a2700c2d605b759748efd2889f5c42d8).